libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -866,11 +866,20 @@ bool ExpressionCompiler::visit(FunctionCall const& _functionCall)
 					else
 					{
 						solAssert(paramTypes[arg - 1]->isValueType(), "");
-						utils().convertType(
-							*arguments[arg - 1]->annotation().type,
-							*paramTypes[arg - 1],
-							true
-						);
+						if (
+							auto functionType =
+							dynamic_cast<FunctionType const*>(arguments[arg - 1]->annotation().type)
```

Comment:
Can you check on the parameter type instead and add an assertion that they are both external function types?

Please also add an assertion that the argument type is not bound.

And maybe also add a test that uses a public library function.